sipXyealink/src/org/sipfoundry/sipxconfig/phone/yealink/YealinkPhone.java

```diff
@@ -215,6 +223,18 @@ public boolean getHasHD() {
     public String getDirectedCallPickupString() {
         return getPhoneContext().getPhoneDefaults().getDirectedCallPickupCode();
     }
+    
+    public String getPhonebookType(){
+        //Setting setting = getSettings().getSetting(PHONEBOOKTYPE);
+        Setting settings = getSettings();
+        Setting setting = settings.getSetting(PHONEBOOKTYPE);
+        String settingValue = (null == setting ? null : (setting.getValue() == null ? null : setting
+                        .getValue()));
+        if (!StringUtils.isEmpty(settingValue)) {
+                    return settingValue;
+                }
+        return "2";
```

Comment:
phonebooktype, is used to generate a 2 types of phonebook xml files. Value 2 - Last name, First Name, Value 1 - First Name, Last name.
If we chose a V7 firmware the parameter is not going to show, value 2 is the default value only used if the parameter is empty.
Value 2 generate phonebook xml with the name Last, First, just like it is now.
About code style, when I have to change the code I will correct that. It's my first time submitting code.